packages/headless/src/controllers/product-listing/facet/headless-product-listing-facet.ts

```diff
@@ -0,0 +1,115 @@
+import {FacetSortCriterion} from '../../../features/facets/facet-set/interfaces/request';
+import {FacetValueState} from '../../../features/facets/facet-api/value';
+import {
+  buildCoreFacet,
+  Facet,
+  FacetOptions,
+  FacetProps,
+  FacetSearch,
+  FacetSearchOptions,
+  FacetSearchState,
+  FacetState,
+  FacetValue,
+  SpecificFacetSearchResult,
+} from '../../core/facets/facet/headless-core-facet';
+import {fetchProductListing} from '../../../features/product-listing/product-listing-actions';
+import {ProductListingEngine} from '../../../app/product-listing-engine/product-listing-engine';
+import {executeToggleFacetSelect} from '../../../features/facets/facet-set/facet-set-controller-actions';
+import {CoreEngine} from '../../../app/engine';
+import {
+  ConfigurationSection,
+  FacetSearchSection,
+  FacetSection,
+  SearchSection,
+} from '../../../state/state-sections';
+import {SearchThunkExtraArguments} from '../../../app/search-thunk-extra-arguments';
+import {loadReducerError} from '../../../utils/errors';
+
+export {
+  FacetOptions,
+  FacetSearchOptions,
+  FacetValueState,
+  FacetProps,
+  Facet,
+  FacetState,
+  FacetSearch,
+  FacetSearchState,
+  SpecificFacetSearchResult,
+  FacetValue,
+};
+
+/**
+ * Creates a `Facet` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `Facet` properties.
+ * @returns A `Facet` controller instance.
+ * */
+export function buildFacet(
+  engine: ProductListingEngine,
+  props: FacetProps
+): Facet {
+  if (!loadFacetReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const {dispatch} = engine;
+  const controller = buildCoreFacet(engine, props);
+  const getFacetId = () => controller.state.facetId;
+
+  return {
+    ...controller,
+
+    toggleSelect: (selection: FacetValue) =>
+      dispatch(
+        executeToggleFacetSelect({
+          facetId: getFacetId(),
+          selection,
+          onAfterSend: (getAnalytics) => {
+            dispatch(fetchProductListing());
+            dispatch(getAnalytics());
```

Comment:
I did this so that the action could be backward compatible.

If someone directly dispatches `executeToggleFacetSelect` right now, it is sending an analytics call. 

So I'd have to duplicate `executeToggleFacetSelect` as a whole just to change one analytics line.

Another solution would be to add an optional parameter `logAnalytics: boolean` and just skip it if set to `false`.